arroyo/processing/strategies/transform.py

```diff
@@ -43,7 +43,7 @@ class ChildProcessTerminated(RuntimeError):
     pass
 
 
-class TransformStep(ProcessingStep[TPayload]):
+class TransformStep(ProcessingStep[TPayload], Generic[TPayload, TTransformed]):
```

Comment:
It did not. It was somehow needed in a previous version that relied on different data structure. Cannto remember anymore why.

Reply:
It was needed because this class depends on two generic parameters: `TPayload` and `TTransformed`. It inherits `TPayload` from the parent class, but it needs to tell the type checker that it has another parameter.

If I do not add it, this part breaks:
https://github.com/getsentry/arroyo/pull/127/files#diff-52195e990ade00fc5326c72366307080a8b5fc04d9ac242a26538679d02bffb1R443-R450

```
next_step: TransformStep[ValuesBatch[str], ValuesBatch[str]] = TransformStep(
```

would be invalid as the type checker expects only one parameter. Thus it should become 

```
next_step: TransformStep[ValuesBatch[str]] = TransformStep(
```

but then the definition of the transformation function would be invalid:

```
def transformer(msg: Message[ValuesBatch[str]],) -> ValuesBatch[str]:
```

Would have to be a 

```
Callable[[ValuesBatch[str]], Nothing]
```

as the code that uses the transformation step does not know about `TTransformed`